.github/workflows/wasm.yml

```diff
@@ -0,0 +1,129 @@
+name: WebAssembly 
+
+on:
+  push:
+    branches: [ main, ci-sandbox ]
+  pull_request:
+    branches: [ '**' ]
+
+env:
+  ccache_basedir: ${{ github.workspace }}
+  ccache_dir: "${{ github.workspace }}/.ccache"
+  ccache_compilercheck: mtime
+  ccache_compress: 'true'
+  ccache_compresslevel: 9
+  ccache_maxsize: 200M
+  ccache_cmake: -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER_LAUNCHER=ccache
+  emsdk_version: 2.0.9
+
+jobs:
+  build-wasm:
+    name: "emscripten"
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Setup Emscripten toolchain
+        run: |
+            git clone --depth 1 https://github.com/emscripten-core/emsdk.git
+            (cd emsdk && ./emsdk install ${{ env.emsdk_version }} ccache-git-emscripten-64bit)
+            (cd emsdk && ./emsdk activate ${{ env.emsdk_version }} ccache-git-emscripten-64bit)
+            # mtime of this file is checked by ccache, we set it to avoid cache misses.
+            touch -m -d '1 Jan 2021 12:00' emsdk/.emscripten
+      - name: Setup Paths
+        run: |
+            # Following two variables are set by emsdk activated otherwise
+            echo "_EMCC_CCACHE=1" >> $GITHUB_ENV
+            echo "EM_CONFIG=$(realpath emsdk/.emscripten)" >> $GITHUB_ENV
+
+            # Configure GitHub Paths
+            echo "${{ github.workspace }}/emsdk" >> $GITHUB_PATH
+            echo "${{ github.workspace }}/emsdk/upstream/emscripten" >> $GITHUB_PATH
+            echo "${{ github.workspace }}/emsdk/node/14.18.2_64bit/bin" >> $GITHUB_PATH
```

Comment:
I expect that path to break at some point when emscripten updates their stuff. In theory node could be one of the tools that makes up an SDK, and so be of a fixed version. But I doubt it is for some reason…

The action I'm using [uses `emsdk construct_env` to print all env variables that the SDK needs, and then exports those selectively to the Github environment](https://github.com/mymindstorm/setup-emsdk/blob/29ba4851d6da084ffdc1e0fc390efadbd75df9d1/src/main.ts#L80-L95).

Reply:
(oops, mixed up the reply fields):

I need to get it to `$GITHUB_PATH` (manually) or use source script in every shell, I think given the existing setting? I like the idea. The example pointed to is JS for consumption in actions which appears to have access to the environment, unsure how to wire the concept in here. Another angle is I can watch pyiodide (which seems to be gaining some traction) and replicate what they're doing. Let me know if you can point me to a concrete way to solve this, happy to pursue it.

On the plus side, such a failure is easily detectable and fixable at which point we can take a second look.

Reply:
I came up with this monstrosity:

```
GITHUB_ENV=github.env
GITHUB_PATH=github.path

rm $GITHUB_ENV $GITHUB_PATH

eval $(./emsdk/emsdk construct_env 2>&1 \
  | grep -v "\+\=" \
  | grep "="  \
  | sed 's/export PATH=\(.*\);/echo \1 >> $GITHUB_PATH;/' \
  | sed 's/export \(.*\);/echo \1 >> $GITHUB_ENV;/' \
  | grep "echo")

tail -v $GITHUB_ENV $GITHUB_PATH
```

I had to additionally add ccache. Unsure how to obtain the entry via `construct_env`. The ccache path looks not pinned to a version, also node-path is now fixed.